src/components/fundamentals/button/index.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { Children } from "react"
+import { classNames } from "../../../utils/class-names"
+import Spinner from "../../atoms/spinner"
+
+type ButtonProps = {
+  variant: "primary" | "secondary" | "ghost"
+  size: "small" | "medium" | "large"
+  loading?: boolean
+} & React.ButtonHTMLAttributes<HTMLButtonElement>
+
+const Button = React.forwardRef<HTMLButtonElement, ButtonProps>(
+  (
+    {
+      variant = "primary",
+      size = "large",
+      loading = false,
+      children,
+      ...attributes
+    },
+    ref
+  ) => {
+    const handleClick = e => {
+      if (!loading && attributes.onClick) {
+        attributes.onClick(e)
+      }
+    }
+
+    let style: string
+
+    switch (variant + "-" + size) {
+      case "primary-large":
+        style = "btn-primary-large"
+        break
+      case "primary-medium":
+        style = "btn-primary-medium"
+        break
+      case "primary-small":
+        style = "btn-primary-small"
+        break
+      case "secondary-large":
+        style = "btn-secondary-large"
+        break
+      case "secondary-medium":
+        style = "btn-secondary-medium"
+        break
+      case "secondary-small":
+        style = "btn-secondary-small"
+        break
+      case "ghost-large":
+        style = "btn-ghost-large"
+        break
+      case "ghost-medium":
+        style = "btn-ghost-medium"
+        break
+      case "ghost-small":
+        style = "btn-ghost-small"
+        break
+      default:
+        style = "btn-primary-large"
+        break
+    }
+
+    return (
+      <button
+        {...attributes}
+        className={classNames(style, attributes.className)}
```

Comment:
tiny nit: maybe a good candidate for using [clsx](https://www.npmjs.com/package/clsx). Current approach is totally valid so feel free to ignore! Just a small heads-up as we'll probably be constructing classes conditionally in the future

Reply:
Are there any significant differences between [`classNames`] and `clsx` @zakariaelas? As far as I can see, they both support conditional classes in an almost identical manner.

Reply:
Correct! As things stand right now, the two would behave the exact same way. Mostly a matter of preference and highly opinionated as personally, i'm not the biggest fan of ternary operators inside `className`.

Only difference could possible be down the line. Consider this example: 

```jsx
// without clsx
const SomeComponent = () => {
  // some state .. using `isOpen` here but could be pretty much anything
  const [isOpen, setIsOpen] = useState(false)
  return (
    <div className={`class-1 class-2 ${isOpen ? "class-open" : ""}`}>
      ...
    </div>
  )
}

// with clsx
const SomeComponent = () => {
  const [isOpen, setIsOpen] = useState(false)
  return (
    <div className={clsx("class-1", "class-2", { "class-open" : isOpen })}>
      ...
    </div>
  )
}
```
No significant difference here ... Just a tiny utility library that would help compose `className` conditionally, especially when having class names depending on multiple pieces of state. Really, feel free to ignore and merge :sweat_smile:

Reply:
I am all in for it! Think it's very nifty and will be very handy in the future 😄

Reply:
Seems super nifty I've replaced the utility method with `clsx` instead. Also removed the switch statement, as the result could be achieved by simply doing `"btn-" + variant + "-" + "size"`. I Tailwind just does not like template literals, since doing `{`btn-${variant}-${size}`}` results in the style not being loaded 😅